regression-test/suites/variant_p0/nested.groovy

```diff
@@ -164,13 +164,64 @@ parallel_pipeline_task_num=7,parallel_fragment_exec_instance_num=4,profile_level
         qt_sql """select v['nested'] from var_nested2 where k < 10 order by k limit 10;"""
-        // explode variant array
+        // 0. nomal explode variant array
         order_qt_explode_sql """select count(),cast(vv['xx'] as int) from var_nested lateral view explode_variant_array(v['nested']) tmp as vv where vv['xx'] = 10 group by cast(vv['xx'] as int)"""
         sql """truncate table var_nested2"""
         sql """insert into var_nested2 values(1119111, '{"eventId":1,"firstName":"Name1","lastName":"Surname1","body":{"phoneNumbers":[{"number":"5550219210","type":"GSM","callLimit":5},{"number":"02124713252","type":"HOME","callLimit":3},{"number":"05550219211","callLimit":2,"type":"WORK"}]}}
 ')"""
         order_qt_explode_sql """select v['eventId'], phone_numbers from var_nested2 lateral view explode_variant_array(v['body']['phoneNumbers']) tmp1 as phone_numbers
 where phone_numbers['type'] = 'GSM' OR phone_numbers['type'] = 'HOME' and phone_numbers['callLimit'] > 2;"""
+
+        // test array_function
+        sql "DROP TABLE IF EXISTS var_nested_array_agg"
+        sql """
+                CREATE TABLE IF NOT EXISTS var_nested_array_agg(
+                    k bigint,
+                    v variant
+                )
+                UNIQUE KEY(`k`)
+                DISTRIBUTED BY HASH(k) BUCKETS 1
+                properties("replication_num" = "1", "disable_auto_compaction" = "false", "enable_unique_key_merge_on_write" = "true", "variant_enable_flatten_nested" = "true");
+            """
+        sql "insert into var_nested_array_agg select * from var_nested"
+        // 1. array_contains
+        qt_sql "select * from var_nested_array_agg where array_contains(cast(v['nested']['xx'] as array<int>), 10) order by k limit 10"
+        // 2. array_agg scalar
+        sql "select k, array_agg(cast(v['nested'] as text))  from var_nested_array_agg group by k limit 10"
```

Comment:
qt_sql ? just sql will not  show the output in out file

Reply:
the output is not stable since the serailized `v['nested']` may contains blank and is not stable， which is acceptable at present